src/Sentry/TransactionTracer.cs

```diff
@@ -253,7 +253,7 @@ public void Finish(Exception exception, SpanStatus status)
             Finish(exception, SpanStatusConverter.FromException(exception));
 
         /// <inheritdoc />
-        public ISpan? GetLastActiveSpan() => Spans.LastOrDefault(s => !s.IsFinished);
+        public ISpan? GetLastActiveSpan() => Spans.OrderByDescending(x => x.StartTimestamp).FirstOrDefault(s => !s.IsFinished);
```

Comment:
```suggestion
        public ISpan? GetLastActiveSpan() => 
            // We need to sort by timestamp because the order of ConcurrentBag<T> is not deterministic
            Spans.OrderByDescending(x => x.StartTimestamp).FirstOrDefault(s => !s.IsFinished);
```